include/xtensor/xmanipulation.hpp

```diff
@@ -847,6 +854,29 @@ namespace xt
         detail::roll(cpy.begin(), e.begin(), shift, saxis, shape, 0);
         return cpy;
     }
+
+    /****************************
+     * repeat implementation    *
+     ****************************/
+    template <class E>
+    inline auto repeat(E&& e, std::ptrdiff_t repeats, std::ptrdiff_t axis)
+    {
+        const auto casted_axis = static_cast<typename std::decay_t<E>::size_type>(axis);
+        std::vector<std::ptrdiff_t> broadcasted_repeats(e.shape(casted_axis));
+        std::fill(broadcasted_repeats.begin(), broadcasted_repeats.end(), repeats);
+        return repeat(std::forward<E>(e), std::forward<std::vector<std::ptrdiff_t>>(broadcasted_repeats), axis);
+    }
+
+    template <class E>
+    inline auto repeat(E&& e, std::vector<std::ptrdiff_t>&& repeats, std::ptrdiff_t axis)
```

Comment:
This API prevents to write something like:

```cpp
std::vector<std::ptrdiff_t> rep = {1, 2, 3};
auto res = xt::repeat(e, rep, 0);
```

An solution will be to add an overload accepting a const reference on the `repeats` vector, and forward the call to an implementation function that accepts a universal reference. In that case, the implementation function should forward the `repeats` arguments when building the `xrepeat` object.